agent-inject/handler.go

```diff
@@ -85,6 +86,14 @@ type Handler struct {
 func (h *Handler) Handle(w http.ResponseWriter, r *http.Request) {
 	h.Log.Info("Request received", "Method", r.Method, "URL", r.URL)
 
+	// Measure request processing duration and monitor request queue
+	requestQueue.Inc()
+	requestStart := time.Now()
+	defer func() {
+		requestProcessingTime.Observe(float64(time.Since(requestStart).Milliseconds()))
```

Comment:
I wonder if would be worth tracking the failures as well. We do something like this in VSO here: https://github.com/hashicorp/vault-secrets-operator/blob/6ab056c22acb5dc4812620233e3b141fb9e02f9c/vault/client.go#L810

Reply:
Makes sense, and it would unblock @tvoran's feedback [here](https://github.com/hashicorp/vault-k8s/pull/703#discussion_r1847772903).

@benashz Given that it goes beyond the scope we were asked for, is it OK if I push that to a follow-on PR?